pkg/project/registry/project/proxy/proxy.go

```diff
@@ -153,20 +160,45 @@ func (s *REST) Update(ctx kapi.Context, name string, objInfo rest.UpdatedObjectI
 	if !ok {
 		return nil, false, fmt.Errorf("not a project: %#v", obj)
 	}
+	namespace := projectutil.ConvertProject(project)
 
 	s.updateStrategy.PrepareForUpdate(ctx, obj, oldObj)
 	if errs := s.updateStrategy.ValidateUpdate(ctx, obj, oldObj); len(errs) > 0 {
-		return nil, false, kerrors.NewInvalid(projectapi.Kind("Project"), project.Name, errs)
+		namespace, err = s.tryImpersonatingNamespaceUpdateOnMetadataError(namespace, ctx, errs)
+	} else {
+		namespace, err = s.client.Update(namespace)
 	}
 
-	namespace, err := s.client.Update(projectutil.ConvertProject(project))
 	if err != nil {
 		return nil, false, err
 	}
 
 	return projectutil.ConvertNamespace(namespace), false, nil
 }
 
+func (s *REST) tryImpersonatingNamespaceUpdateOnMetadataError(namespace *kapi.Namespace, ctx kapi.Context, errs field.ErrorList) (*kapi.Namespace, error) {
+	// If the error is in the metadata's annotations or labels, we try to impersonate the user and update the namespace
+	if containsAnnotationOrLabelMetadataError(errs) {
```

Comment:
This decision should be made before the method call.  Maybe a switch/case after checking the strategy.